sabnzbd/nzbqueue.py

```diff
@@ -580,8 +580,13 @@ def sort_by_size(self, reverse: bool = False):
         logging.info("Sorting by size... (reversed: %s)", reverse)
         self.__nzo_list = sort_queue_function(self.__nzo_list, _nzo_size_cmp, reverse)
 
+    @NzbQueueLocker
+    def sort_by_completed(self, reverse: bool = True):
+        logging.debug("Sorting by completed...")
+        self.__nzo_list = sort_queue_function(self.__nzo_list, _nzo_completed_cmp, reverse)
```

Comment:
I also worry about it just randomly sorting jobs at the bottom that have 0% completion. Did you check that?
Otherwise we should only apply the sort on jobs with >0%.

Reply:
It shouldn't happen.

> The [sort()](https://docs.python.org/3/library/stdtypes.html#list.sort) method is guaranteed to be stable. A sort is stable if it guarantees not to change the relative order of elements that compare equal [...]